proposals/NNNN-migration-tooling-for-upcoming-features.md

```diff
@@ -0,0 +1,124 @@
+# Migration tooling for Swift upcoming features
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: TBD
+* Review: TBD
+
+## Introduction
+
+As Swift evolves, source-breaking changes to the language are staged behind new
+language modes and, more recently, [upcoming features][SE-0362] that may be
+enabled piecemeal without adopting an entire new language mode. 
+
+In this proposal, we zoom in on the experience of adopting *individual* 
+source-breaking upcoming features. The premise is that, in principle, migration 
+is generally separable from adoption (?), and that decoupling migration will 
+empower compiler engineers to offer significantly richer diagnostics and 
```

Comment:
The focus here should be on what this enables for programmers, not compiler engineers (e.g. this mode allows programmers to have specific diagnostics on where their code needs to change to preserve existing behavior -- guidance that is not appropriate for a regular build)